Improve image transcription

baconbm opened the pull request:
This PR utilizes OpenCV to improve the image transcription process. It enhances images by enlarging them slightly, making it easier for the OCR library to see the text within the image. Closes #27

afig commented:
I think sticking with just the image resizing for now is a good idea.

Is there any particular reason you chose to use OpenCV to do the image resizing? Pillow, the fork of PIL (Python Image Library) that we are already using seems to have an [image resize feature](https://pillow.readthedocs.io/en/3.1.x/reference/Image.html#PIL.Image.Image.resize) as well.

Unless we need OpenCV's implementation for something, we may want to use Pillow/PIL's implementation to avoid a new dependency. (OpenCV is also a huge library whose main purpose is to do something completely different, and so using it for just image resizing seems misleading and inefficient).

Also, it seems like the unit tests for ExtractURL are being deleting in this PR (`testExtractURL.py`). Is this intentional?

baconbm commented:
I agree @afig, while cv2 does have some cool features within it, importing such a big library for a small implementation is overkill. Perhaps we can find some use for it in the future, but for the current processing it is unnecessary. I replaced it with the `resize` method provided by `PIL`. 

The `testExtractURL.py` was the testing file I was using before we implemented unit testing, and the function that was being tested within it was made redundant by the unit tests that were implemented later, so I removed it within this PR.

afig commented on src/TextifyBot.py:
```diff
@@ -9,7 +9,9 @@
 import pytesseract
-import time
+from PIL import Image
+Import time
```
```suggestion
import time
```

afig commented on src/TextifyBot.py:
```diff
@@ -74,12 +76,23 @@ def allowedToParse(postID):
     return pytesseract.image_to_string(image)
 
 
+# converts downloaded image to grayscale
```
```suggestion
```

afig commented on src/TextifyBot.py:
```diff
@@ -74,12 +76,23 @@ def allowedToParse(postID):
 
 
+# converts downloaded image to grayscale
+# and converts all pixels close to black to full black / all pixels close to white to full white
```
```suggestion
# Makes the given image larger to improve success of transcription
```
